### PR TITLE
Add hydra run directory, change READMEs, pull recent code from inner source

### PR DESCRIPTION
**Pull request by liranszlak**
This PR contains:
- a run directory with basic yamls for different use cases (not exhustive, we might want to add additional yamls in the future)
- pulled code from inner source with gradient reversal capabilities
- changes to READMEs in the repo

### CODE REVIEW

File: run/fine_tune_train_and_test_config.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+seed:
+  seed_value: 1234
+
+model:
+  checkpoint: null
+
+data_module:
+  max_length: 32
+  data_dir: ${oc.env:BMFM_TARGETS_ZHENG68K_DATA}
```
Comment (mmdanziger) [inline run/fine_tune_train_and_test_config.yaml:19]: this should be in the zheng68k yaml not here

File: run/fine_tune_train_config.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+model:
+  checkpoint: null
+  hidden_size: 16
+  intermediate_size: 32
+  num_hidden_layers: 2
+  num_attention_heads: 2
+
+data_module:
+  max_length: 32
+  data_dir: ${oc.env:BMFM_TARGETS_ZHENG68K_DATA}
```
Comment (mmdanziger) [inline run/fine_tune_train_config.yaml:23]: move to data module yaml

File: run/perturbation_config.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+seed:
+  seed_value: 1234
+
+model:
+  checkpoint: null
+
+data_module:
+  max_length: 32
+  data_dir: /perturbation_data/
```
Comment (mmdanziger) [inline run/perturbation_config.yaml:18]: move to data module yaml, set to env var

File: bmfm_targets/README.md [hunk 4]
```diff
@@ -58,50 +31,6 @@ Currently there are 10 finetuning data that we are running on our pre-trained dn
 
 The configuration is structured into different sections to define various aspects of the experiment.
 
-By default the app makes use of the following environment variables. Below are their settings for CCC.
-To use the scripts either copy these lines into your bashrc or override them in your config yaml.
-
-```bash
-export BMFM_TARGETS_PANGLAO_DATA=/dccstor/bmfm-targets/data/omics/transcriptome/scRNA/pretrain/panglao/
-export BMFM_TARGETS_PANGLAO_METADATA=/dccstor/bmfm-targets/data/omics/transcriptome/scRNA/pretrain/panglao/metadata/metadata.csv
-export BMFM_TARGETS_ZHENG68K_DATA=/dccstor/bmfm-targets/data/omics/transcriptome/scRNA/finetune/Zheng68K/
```
Comment (mmdanziger) [inline bmfm_targets/README.md:67]: you're still using this so you need to set it to something, maybe give instructions on how to get the data

File: bmfm_targets/README.md [hunk 5]
```diff
@@ -305,15 +234,3 @@ During testing, three types of CI for the evaluation metrics are available thoru
 ```jbsub -mem 32g -cores 1+1 -q x86_6h -require a100 bmfm-targets-scbert -cn scbert_train_panglaodb```
 
 Note that the environment variables (as detailed in the Configuration section above) need to be exported within the user .bashrc file (located in the CCC user root directory `/u/{user_name}/.bashrc`).
```
Comment (mmdanziger) [inline bmfm_targets/README.md:236]: remove

**Review by mmdanziger — CHANGES_REQUESTED** [review]
see notes on cleaning up the yamls. we'll be working on the readme more today

File: run/fine_tune_train_and_test_config.yaml (hunk 1 above)
Comment (liranszlak) [reply run/fine_tune_train_and_test_config.yaml:19]: why? I want people to have this option clearly visible in the yaml template so that if they need to change it they know how to, if it's hidden in an inner yaml it's harder to change (it's here for the example, to show how you would change it if need be)

File: run/fine_tune_train_config.yaml (hunk 2 above)
Comment (liranszlak) [reply run/fine_tune_train_config.yaml:23]: I guess it's the same explanation as the previous comment - if you think it should be moved I'll move it, just thought it's better to have it visible to external users for ease of usage

File: bmfm_targets/README.md (hunk 4 above)
Comment (liranszlak) [reply bmfm_targets/README.md:67]: I thought we would remove the use of this entirely - what do you think? since there is no necessity for it, things can run when people just set the data_dir to wherever they saved their data

Comment (mmdanziger) [reply bmfm_targets/README.md:67]: if you do not reference it in the yamls then it's fine

**Review by mmdanziger — APPROVED** [review]
Looks good to me, there's more work to be done but let's merge and iterate.
